tests/MongoDB.Bson.Tests/ObjectModel/ObjectIdTests.cs

```diff
@@ -14,20 +14,76 @@
 */
 
 using System;
+using System.Globalization;
 using System.Linq;
-using MongoDB.Bson;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers;
 using Xunit;
 
 namespace MongoDB.Bson.Tests
 {
     public class ObjectIdTests
     {
+        [Theory]
+        [InlineData(0xFFFFFE, 0xFFFFFF)]
+        [InlineData(0xFFFFFF, 0)]
+        [InlineData(0x1000000, 1)]
+        [InlineData(0x1FFFFFE, 0xFFFFFF)]
+        [InlineData(0x1FFFFFF, 0)]
+        [InlineData(0x2000000, 1)]
+        public void Ensure_that_counter_is_resetted_after_max_value(int incrementValue, int expectedIncrement)
+        {
+            ObjectIdReflector.__staticIncrement(incrementValue);
+            var oid = ObjectId.GenerateNewId();
+            var resultCounter = oid.Increment;
+            resultCounter.Should().Be(expectedIncrement);
+        }
```

Comment:
I prefer to say "wraps around" than "reset". This test can also be slightly simplified:

    public void Ensure_that_increment_wraps_around_after_max_value(int seedIncrement, int expectedIncrement)
    {
        ObjectIdReflector.__staticIncrement(seedIncrement);
        var objectId = ObjectId.GenerateNewId();
        objectId.Increment.Should().Be(expectedIncrement);
    }

Nice idea for how to test this!

Reply:
Naming nit: we avoid abbreviations, so prefer "objectId" to "oid".